File: test/data/phenology/crop2ml/algo/java/UpdateLeafFlag.java
if (phase >= 1.0 && phase< 4.0)
{
    if (leafNumber > 0.0)
    {
        if (hasFlagLeafLiguleAppeared == 0 && (finalLeafNumber > 0 && leafNumber >= finalLeafNumber))
        {
            hasFlagLeafLiguleAppeared = 1;
            if  (calendarMoments.contains("FlagLeafLiguleJustVisible")==false)
            {
                calendarMoments.add("FlagLeafLiguleJustVisible");
                calendarCumuls.add(cumulTT);
                calendarDates.add(currentdate);
            }
        }
    }
    else
        hasFlagLeafLiguleAppeared = 0;
}


    

File: test/data/phenology/crop2ml/algo/cpp/UpdateLeafFlag.cpp

if (phase >= 1 && phase < 4)
{
    if (leafNumber > 0)
    {
        if (hasFlagLeafLiguleAppeared == 0 && (finalLeafNumber > 0 && leafNumber >= finalLeafNumber))
        {
            hasFlagLeafLiguleAppeared = 1;
            size_t pos = find(calendarMoments.begin(), calendarMoments.end(), "FlagLeafLiguleJustVisible") - calendarMoments.begin();

            if (pos >= calendarMoments.size())
            {
                calendarMoments.push_back("FlagLeafLiguleJustVisible");
                calendarCumuls.push_back(cumulTT);
                calendarDates.push_back(currentdate);
            }
        }
    }
    else
        hasFlagLeafLiguleAppeared = 0;
}
